### PR TITLE
PLATFORM-790 Create Revision Resource in Pandora Service

### PR DESCRIPTION
**Pull request by sqreek**

### CODE REVIEW

File: pandora-service/src/main/java/com/wikia/pandora/api/service/RevisionService.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.wikia.pandora.api.service;
+
+import com.wikia.mwapi.domain.Page;
+import com.wikia.pandora.domain.Revision;
+
+public interface RevisionService {
```
Comment (pchojnacki) [inline pandora-service/src/main/java/com/wikia/pandora/api/service/RevisionService.java]: We talked about this already. Having services as interface doesn't make much sense.

This is 1:1 relationship please remove this interface and only use concrete class

File: pandora-service/src/main/java/com/wikia/pandora/service/ServiceFactory.java [hunk 2]
```diff
@@ -24,6 +25,8 @@ public ServiceFactory(PandoraConfiguration configuration, Environment environmen
   public abstract CategoryService createCategoryService();
 
   public abstract CommentService createCommentService();
+
```
Comment (pchojnacki) [inline pandora-service/src/main/java/com/wikia/pandora/service/ServiceFactory.java]: pease dont use this Factory its unnecessary

File: pandora-service/src/main/java/com/wikia/pandora/service/mediawiki/MediawikiRevisionService.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package com.wikia.pandora.service.mediawiki;
+
+import com.wikia.mwapi.domain.ApiResponse;
+import com.wikia.mwapi.domain.Page;
+import com.wikia.pandora.api.service.RevisionService;
+import com.wikia.pandora.domain.Revision;
+import com.wikia.pandora.domain.builder.RevisionBuilder;
+import com.wikia.pandora.gateway.mediawiki.MediawikiGateway;
+
+public class MediawikiRevisionService extends MediawikiService implements RevisionService {
```
Comment (pchojnacki) [inline pandora-service/src/main/java/com/wikia/pandora/service/mediawiki/MediawikiRevisionService.java]: Skip the implements part

File: pandora-service/src/main/java/com/wikia/pandora/service/mediawiki/MediawikiServiceFactory.java [hunk 4]
```diff
@@ -43,4 +44,9 @@ public CommentService createCommentService() {
 
     throw new NotImplementedException("");
   }
+
+  @Override
+  public RevisionService createRevisionService() {
```
Comment (pchojnacki) [inline pandora-service/src/main/java/com/wikia/pandora/service/mediawiki/MediawikiServiceFactory.java]: unnecessary

File: pandora-service/src/main/java/com/wikia/pandora/service/mercury/MercuryServiceFactory.java [hunk 5]
```diff
@@ -43,6 +44,11 @@ public CommentService createCommentService() {
     throw new NotImplementedException("");
   }
 
+  @Override
+  public RevisionService createRevisionService() {
```
Comment (pchojnacki) [inline pandora-service/src/main/java/com/wikia/pandora/service/mercury/MercuryServiceFactory.java]: unnecessary

**Comment by pchojnacki** [comment]
Once you get rid of this unecessary interface then this is good to merge.

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by sqreek** [comment]
I will remove fabric in separate pull request, becuse now, it will provide conflict with 
https://github.com/Wikia/pandora/pull/32
https://github.com/Wikia/pandora/pull/25

**Comment by pchojnacki** [comment]
ok

**Pull request merged**